src/psyclone/tests/domain/lfric/lfric_extract_driver_creator_test.py

```diff
@@ -0,0 +1,510 @@
+                                          proxy_name_mapping={})
+    assert ("Unexpected type 'str' in _flatten_reference, it must be a "
+            "'StructureReference'" in str(err.value))
+
+
+# ----------------------------------------------------------------------------
+def test_lfric_driver_add_call(fortran_writer):
+    '''Tests that adding a call detects errors and adds calls
+    with and without parameters as expected.
+    '''
+    program = Routine("routine", is_program=True)
+    program.symbol_table.find_or_create_tag("test")
+    driver_creator = LFRicExtractDriverCreator()
+    with pytest.raises(TypeError) as err:
+        driver_creator._add_call(program, "test", [])
+    assert ("Routine 'test' is a symbol of type 'Symbol', not a "
+            "'RoutineSymbol'" in str(err.value))
+
+    driver_creator._add_call(program, "my_sub", [])
+    driver_creator._add_call(program, "my_sub_2", [Literal("1", INTEGER_TYPE)])
+    out = fortran_writer(program)
+    assert "call my_sub()" in out
+    assert "call my_sub_2(1)" in out
+
+
+# ----------------------------------------------------------------------------
+def test_lfric_driver_import_modules():
+    '''Tests that adding a call detects errors as expected.
+    '''
+    program = Routine("routine", is_program=True)
+    _, invoke = get_invoke("8_vector_field_2.f90", API,
+                           dist_mem=False, idx=0)
+
+    sched = invoke.schedule
+    # We need to lower to convert the kernels to calls
+    sched.lower_to_language_level()
+
+    driver_creator = LFRicExtractDriverCreator()
+    assert ["routine"] == [sym.name for sym in program.symbol_table.symbols]
+
+    driver_creator._import_modules(program, sched)
+    # We should now have two more symbols:
+    all_symbols = ["routine", "testkern_coord_w0_2_mod",
+                   "testkern_coord_w0_2_code"]
+    assert (all_symbols == [sym.name for sym in program.symbol_table.symbols])
+
+    # Import twice so we test the handling of symbols that
+    # are already in the symbol table:
+    driver_creator._import_modules(program, sched)
+
+    # The symbol table should be the same as it was before:
+    assert (all_symbols == [sym.name for sym in program.symbol_table.symbols])
+
+
+# ----------------------------------------------------------------------------
+def test_lfric_driver_import_modules_no_import_interface(fortran_reader):
+    '''This test checks the import_modules method if there is a call
+    that has no ImportInterface by calling an unknown function.'''
+
+    code = """subroutine test()
+              call something()
+              end subroutine"""
+
+    psyir = fortran_reader.psyir_from_source(code)
+    # Find the schedule:
+    sched = psyir.walk(Schedule)[0]
+    sched.lower_to_language_level()
+    driver_creator = LFRicExtractDriverCreator()
+    program = Routine("routine", is_program=True)
+    driver_creator._import_modules(program, sched)
```

Comment:
While this is also old-style, it would also allow us to handle:
```
call flush(1)
```
Yes, it's now the user's problem to link it all together ... but that's the case anyway :)  

Do we officially have a statement saying that PSyclone only supports calls of subroutines that are either local to the module or (explicitly??) imported? I would actually prefer to leave it this way, I can't see any harm (except not aborting on bad coding practices :)  ).

Reply:
OK, we'll leave as is. (The Fortran backend has recently been extended/fixed to allow this situation so this change is consistent with that.)